proto/redpanda/api/dataplane/v1alpha1/pipeline.proto

```diff
@@ -0,0 +1,212 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/field_mask.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+message Pipeline {
+  string id = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.len = 20
+  ];
+
+  string display_name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+    (buf.validate.field).string.min_len = 3,
+    (buf.validate.field).string.max_len = 128
+  ];
+
+  string description = 3 [(buf.validate.field).string.max_len = 1000];
```

Comment:
Being curious: Where do we store this in the case of cloud? I imagine we have some limits there. Possibly worth to restricting it to less characters (256 or so)? We need to find something that supported across all our cloud providers.

Reply:
Good point. This will be stored in k8s